pulser-core/pulser/noise_model.py

```diff
@@ -257,9 +260,14 @@ def to_tuple(obj: tuple) -> tuple:
 
         # Get rid of unnecessary None's
         for p_, val in param_vals.items():
-            param_vals[p_] = (
-                self._get_default_value(p_) if val is None else val
-            )
+            if p_ in _PROBABILITY_LIKE | _POSITIVE and not isinstance(
+                val, (int, float)
+            ):
+                raise TypeError(
+                    f"Value for {p_} should be positive, "
+                    f"{'between 0 and 1, ' if p_ in _PROBABILITY_LIKE else ''}"
+                    f"not {val}."
+                )
```

Comment:
The value is being checked in `_validate_parameters()`, I think the message here should stick to the type check